File: minute/src/lexer.ts
import { Token } from "./models/token";

const TEXT = "text";
const STRONG = "strong";

const STRONG_ELM_REGXP = /\*\*(.*?)\*\*/;

const genTextElement = (id: number, text: string, parent: Token): Token => {
  return {
    id,
    elmType: TEXT,
    content: text,
    parent,
  };
};

const genStrongElement = (id: number, text: string, parent: Token): Token => {
  return {
    id,
    elmType: STRONG,
    content: "",
    parent,
  };
};

const matchWithStrongRegxp = (text: string) => {
  const result: RegExpMatchArray | null = text.match(STRONG_ELM_REGXP);
  console.log("this is matched: ", result);
  return result;
};

export { genTextElement, genStrongElement, matchWithStrongRegxp };

File: 33114098/markdown/lexer.c
#include <stdlib.h>

#include "../include/markdown.h"
#include "../include/token.h"

const char TEXT[] = "text";
const char STRONG[] = "strong";

const char STRONG_ELM_REGEXP[] = "\\*\\*(.*?)\\*\\*";

Token_t *gen_text_elem(int id, char *text, Token_t *parent) {
  Token_t *token = NULL;
  token = (Token_t *)malloc(sizeof(Token_t));

  token->id = id;
  token->parent = parent;
  token->elm_type = TEXT;
  token->content = text;

  return token;
}

Token_t *gen_strong_elem(int id, char *text, Token_t *parent) {
  Token_t *token = NULL;
  token = (Token_t *)malloc(sizeof(Token_t));

  token->id = id;
  token->parent = parent;
  token->elm_type = STRONG;
  token->content = text;

  return token;
}
